docs/source/faq.rst

```diff
@@ -425,3 +425,71 @@ This method is justified in the following way:
 First, if we apply the transformation :math:`x = - \log (u)` to the variable :math:`u` sampled from the uniform distribution :math:`Uni(0, 1)` in the interval :math:`[0, 1]`, the variable :math:`x` will follow the exponential distribution :math:`Exp(1)` with scale parameter :math:`1`.
 Furthermore, for :math:`n` variables :math:`x[0], ..., x[n-1]` that follow the exponential distribution of scale parameter :math:`1` independently, normalizing them with :math:`p[i] = x[i] / \sum_i x[i]`, the vector :math:`p` follows the Dirichlet distribution :math:`Dir(\alpha)` of scale parameter :math:`\alpha = (1, ..., 1)`.
 You can verify the transformation by calculating the elements of the Jacobian.
+
+How can I optimize a model with some constraints?
+-------------------------------------------------
+
+When you want to optimize a model with constraints, you can use the following classes, :class:`~optuna.integration.BoTorchSampler` or :class:`~optuna.samplers.NSGAIISampler`.
+An example using the former class is `this <https://github.com/optuna/optuna-examples/blob/main/multi_objective/botorch_simple.py>`_. This example is a benchmark of Binh and Korn function, a multi-objective optimization, with constraints. This one has two constraints :math:`c0 = (x-5)^2 + y^2 - 25` and :math:`c1 = -((x - 8)^2) - (y + 3)^2 + 7.7` and find the optimal solution satisfying these conditions.
+
+This is the example that uses the latter one.
+
+.. code-block:: python
+
+    import optuna
+
+    def objective(trial):
+        # Binh and Korn function with constraints.
+        x = trial.suggest_float("x", -15, 30)
+        y = trial.suggest_float("y", -15, 30)
+
+        # Constraints which are considered feasible if less than or equal to zero.
+        # The feasible region is basically the intersection of a circle centered at (x=5, y=0)
+        # and the complement to a circle centered at (x=8, y=-3).
+        c0 = (x - 5) ** 2 + y ** 2 - 25
+        c1 = -((x - 8) ** 2) - (y + 3) ** 2 + 7.7
+
+        # Store the constraints as user attributes so that they can be restored after optimization.
+        trial.set_user_attr("constraint", (c0, c1))
+
+        v0 = 4 * x ** 2 + 4 * y ** 2
+        v1 = (x - 5) ** 2 + (y - 5) ** 2
+
+        return v0, v1
+
+
+    def constraints(trial):
+        return trial.user_attrs["constraint"]
+
+
+    if __name__ == "__main__":
```

Comment:
Is this necessary?

Reply:
No, I don't think so.
I followed it because the sample file has this statement, but I should follow examples in this FAQ.